checks/http.go

```diff
@@ -33,8 +33,13 @@ type HTTPCheckConfig struct {
 	Client *http.Client
 	// Timeout is the timeout used for the HTTP request, defaults to "1s".
 	Timeout time.Duration
+	// Options allow you to configure the HTTP request with arbitrary settings, e.g. add request headers, etc.
+	Options []RequestOption
 }
 
+// RequestOption configures the request with arbitrary settings, e.g. add request headers, etc.
```

Comment:
Isn't this too permissive? If you get a request at your hand you can configure it, but you can do more. E.g. execute it right?...
I'm not sure (didn't check) but is there a way to make it less permissive?

Reply:
Request is a struct, and people need to access req.Header to modify the headers, so I don't think I can extract a restrictive interface unless I totally wrap the request or let the user create one from scratch. Got any concrete ideas here?

Reply:
You could wrap is, but that's a lot of work, nevermind